File: mrcomic-theme-platform/src/models/Download.js
const mongoose = require('mongoose');

const downloadSchema = new mongoose.Schema({
  themeId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Theme',
    required: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: false // Может быть анонимная загрузка
  },
  deviceInfo: {
    model: String,
    os: String,
    appVersion: String
  },
  downloadedAt: {
    type: Date,
    default: Date.now
  }
});

// Индексы для аналитики
downloadSchema.index({ themeId: 1 });
downloadSchema.index({ downloadedAt: -1 });
downloadSchema.index({ 'deviceInfo.os': 1 });

const Download = mongoose.model('Download', downloadSchema);

module.exports = Download;

File: app/src/main/java/com/example/mrcomic/theme/data/model/Download.kt
package com.example.mrcomic.theme.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

/**
 * Модель данных для загрузки темы
 */
@Entity(tableName = "downloads")
data class Download(
    @PrimaryKey
    @SerializedName("id")
    val id: String,
    
    @SerializedName("themeId")
    val themeId: String,
    
    @SerializedName("userId")
    val userId: String?,
    
    @SerializedName("deviceInfo")
    val deviceInfo: DeviceInfo,
    
    @SerializedName("downloadedAt")
    val downloadedAt: String
)

data class DeviceInfo(
    @SerializedName("model")
    val model: String,
    
    @SerializedName("os")
    val os: String,
    
    @SerializedName("appVersion")
    val appVersion: String
)
